src/Security/OpenIdConfigurationProvider.php

```diff
@@ -192,10 +197,32 @@ public function getAuthorizationUrl(array $options = []): string
      *
      * @throws ItkOpenIdConnectException
      */
-    public function validateIdToken(string $idToken, string $nonce): object
+
+    /**
+     * Do any required verification of the id token and return an array of decoded claims
+     *
+     * @param string $idToken
+     *   Raw id token as string
+     *
+     * @param string $nonce
+     *   Nonce as string
+     *
+     * @param int $leeway
+     *   Leeway set in seconds. Defaults to 0 and must be positive
+     *
+     * @return object
+     *
+     * @throws ItkOpenIdConnectException
+     */
+    public function validateIdToken(string $idToken, string $nonce, int $leeway = 0): object
     {
+        if ($leeway < 0) {
+            throw new \InvalidArgumentException('$leeway has to be a positive integer');
```

Comment:
Please don't use PHP's built-in exceptions directly. USe one of the exceptions in `src/Exception` or introduce a new one if needed.

The idea is that all exceptions thrown extend `ItkOpenIdConnectException`. That way a user of the lib can decide wheater to just catch `ItkOpenIdConnectException` or catch the specific exception.

....This properly deserves an entry in the readme.

Reply:
Also this is really easy to write at test for - just saying 😃

Reply:
Fixed in 4804dbd, b49693c and in 81e595b.